### PR TITLE
Allow Manufacturer's code to be set from constants

### PR DESCRIPTION
**Pull request by MartinDaCosta53**

### CODE REVIEW

File: examples/VLCB_1in1out/VLCB_1in1out.ino [hunk 1]
```diff
@@ -1,5 +1,4 @@
 //  Copyright (C) Sven Rosvall (sven@rosvall.ie)
-//  Copyright (C) Sven Rosvall (sven@rosvall.ie)
 //  This file is part of VLCB-Arduino project on https://github.com/SvenRosvall/VLCB-Arduino
```
Comment (SvenRosvall) [inline examples/VLCB_1in1out/VLCB_1in1out.ino:2]: Oops!

File: examples/VLCB_1in1out/VLCB_1in1out.ino [hunk 2]
```diff
@@ -1,5 +1,4 @@
 //  Copyright (C) Sven Rosvall (sven@rosvall.ie)
-//  Copyright (C) Sven Rosvall (sven@rosvall.ie)
```
Comment (SvenRosvall) [inline examples/VLCB_1in1out/VLCB_1in1out.ino:2]: Oops!

File: examples/VLCB_1in1out/VLCB_1in1out.ino [hunk 3]
```diff
@@ -46,16 +46,16 @@ const byte SWITCH0 = 8;             // VLCB push button switch pin
 VLCB::Configuration modconfig;               // configuration object
 VLCB::CAN2515 can2515;                  // CAN transport object
 VLCB::LEDUserInterface ledUserInterface(LED_GRN, LED_YLW, SWITCH0);
-VLCB::SerialUserInterface serialUserInterface(&can2515);
+VLCB::SerialUserInterface serialUserInterface(&modconfig, &can2515);
+VLCB::CombinedUserInterface combinedUserInterface(&ledUserInterface, &serialUserInterface);
```
Comment (SvenRosvall) [inline examples/VLCB_1in1out/VLCB_1in1out.ino:50]: I think you have your repo out of sync again. The CombinedUserInterface is gone. 
Line 49 is also out of date.

File: examples/VLCB_1in1out/VLCB_1in1out.ino [hunk 4]
```diff
@@ -46,16 +46,16 @@ const byte SWITCH0 = 8;             // VLCB push button switch pin
...
 VLCB::CanService canService(&can2515);
 VLCB::NodeVariableService nvService;
-VLCB::ConsumeOwnEventsService coeService;
 VLCB::EventConsumerService ecService;
 VLCB::EventTeachingService etService;
 VLCB::EventProducerService epService;
-VLCB::Controller controller(&modconfig,
-                            {&mnService, &ledUserInterface, &serialUserInterface, &canService, &nvService, &ecService, &epService, &etService, &coeService}); // Controller object
+VLCB::Controller controller(&combinedUserInterface, &modconfig, &can2515, 
+                            { &mnService, &canService, &nvService, &ecService, &epService, &etService }); // Controller object
```
Comment (SvenRosvall) [inline examples/VLCB_1in1out/VLCB_1in1out.ino:58]: As above, this out of sync.

File: examples/VLCB_1in1out/VLCB_1in1out.ino [hunk 5]
```diff
@@ -65,7 +65,7 @@ VLCB::LED moduleLED(6);                  // an example LED as output
 unsigned char mname[7] = { '1', 'I', 'N', '1', 'O', 'U', 'T' };
 
 // forward function declarations
-void eventhandler(byte, const VLCB::VlcbMessage *);
+void eventhandler(byte, VLCB::VlcbMessage *);
```
Comment (SvenRosvall) [inline examples/VLCB_1in1out/VLCB_1in1out.ino:68]: Out of sync.

File: examples/VLCB_4in4out/VLCB_4in4out.ino [hunk 6]
```diff
@@ -77,7 +77,8 @@
 const byte VER_MAJ = 1;             // code major version
 const char VER_MIN = 'a';           // code minor version
 const byte VER_BETA = 0;            // code beta sub-version
-const byte MODULE_ID = 98;          // VLCB module type
+const byte MODULE_ID = 82;          // VLCB module type
+const byte MANUFACTURER = MANU_DEV; // for boards in development.
```
Comment (SvenRosvall) [inline examples/VLCB_4in4out/VLCB_4in4out.ino:81]: Pleas set manufacturer before module ID. 
No functional difference. Just looks nicer as manufacturer has higher precedence.

File: examples/VLCB_4in4out/VLCB_4in4out.ino [hunk 7]
```diff
@@ -143,8 +144,8 @@ void setupVLCB()
   // set module parameters
   VLCB::Parameters params(modconfig);
   params.setVersion(VER_MAJ, VER_MIN, VER_BETA);
-  params.setManufacturer(MANU_DEV);
   params.setModuleId(MODULE_ID);
+  params.setManufacturer(MANUFACTURER);
```
Comment (SvenRosvall) [inline examples/VLCB_4in4out/VLCB_4in4out.ino:148]: Pleas set manufacturer before module ID. 
No functional difference. Just looks nicer as manufacturer has higher precedence.

File: examples/VLCB_empty/VLCB_empty.ino [hunk 8]
```diff
@@ -47,14 +48,14 @@ VLCB::NodeVariableService nvService;
 VLCB::EventConsumerService ecService;
 VLCB::EventTeachingService etService;
 VLCB::EventProducerService epService;
-VLCB::Controller controller(&modconfig,
-                            {&mnService, &userInterface, &canService, &nvService, &ecService, &epService, &etService}); // Controller object
+VLCB::Controller controller(&userInterface, &modconfig, &can2515, 
+                            { &mnService, &canService, &nvService, &ecService, &epService, &etService }); // Controller object
```
Comment (SvenRosvall) [inline examples/VLCB_empty/VLCB_empty.ino:52]: Out of sync as described above.

File: examples/VLCB_4in4out/VLCB_4in4out.ino (hunk 7 above)
Comment (MartinDaCosta53) [reply examples/VLCB_4in4out/VLCB_4in4out.ino:147]: Done.  I've also modified in empty and 1in1out. You're right about precedence.  I'm sure that I thought about this and then got it wrong!  I blame chemo!

File: examples/VLCB_1in1out/VLCB_1in1out.ino [hunk 9]
```diff
@@ -50,12 +49,11 @@ VLCB::SerialUserInterface serialUserInterface(&can2515);
...
 VLCB::CanService canService(&can2515);
 VLCB::NodeVariableService nvService;
-VLCB::ConsumeOwnEventsService coeService;
 VLCB::EventConsumerService ecService;
 VLCB::EventTeachingService etService;
 VLCB::EventProducerService epService;
-VLCB::Controller controller(&modconfig,
-                            {&mnService, &ledUserInterface, &serialUserInterface, &canService, &nvService, &ecService, &epService, &etService, &coeService}); // Controller object
+VLCB::Controller controller(&modconfig, 
+                            { &mnService, &ledUserInterface, &serialUserInterface, &canService, &nvService, &ecService, &epService, &etService }); // Controller object
```
Comment (SvenRosvall) [inline examples/VLCB_1in1out/VLCB_1in1out.ino:56]: Why have you taken out coeService? 
I actually use it for my own testing,

File: examples/VLCB_empty/VLCB_empty.ino [hunk 10]
```diff
@@ -54,7 +55,7 @@ VLCB::Controller controller(&modconfig,
 unsigned char mname[7] = { 'E', 'M', 'P', 'T', 'Y', ' ', ' ' };
 
 // forward function declarations
-void eventhandler(byte, const VLCB::VlcbMessage *);
+void eventhandler(byte, VLCB::VlcbMessage *);
```
Comment (SvenRosvall) [inline examples/VLCB_empty/VLCB_empty.ino:58]: Still need the const here.

File: examples/VLCB_empty/VLCB_empty.ino [hunk 11]
```diff
@@ -172,7 +173,7 @@ void loop()
 /// called from the VLCB library when a learned event is received
 /// it receives the event table index and the CAN frame
 //
-void eventhandler(byte index, const VLCB::VlcbMessage *msg)
+void eventhandler(byte index, VLCB::VlcbMessage *msg)
```
Comment (SvenRosvall) [inline examples/VLCB_empty/VLCB_empty.ino:176]: Still need the const here.

File: examples/VLCB_1in1out/VLCB_1in1out.ino (hunk 9 above)
Comment (MartinDaCosta53) [reply examples/VLCB_1in1out/VLCB_1in1out.ino:56]: Maybe I'm not as with it today as I thought I was.  I'll amend with anything else from the comments.

File: examples/VLCB_1in1out/VLCB_1in1out.ino [hunk 12]
```diff
@@ -216,7 +214,7 @@ void processModuleSwitchChange()
 /// called from the VLCB library when a learned event is received
 /// it receives the event table index and the CAN frame
 //
-void eventhandler(byte index, const VLCB::VlcbMessage *msg)
+void eventhandler(byte index, VLCB::VlcbMessage *msg)
```
[the remark this answers is not in the document]
Comment (MartinDaCosta53) [reply examples/VLCB_1in1out/VLCB_1in1out.ino]: Done

File: examples/VLCB_empty/VLCB_empty.ino (hunk 10 above)
Comment (MartinDaCosta53) [reply examples/VLCB_empty/VLCB_empty.ino]: Done

File: examples/VLCB_empty/VLCB_empty.ino (hunk 11 above)
Comment (MartinDaCosta53) [reply examples/VLCB_empty/VLCB_empty.ino]: Done

**Pull request merged**